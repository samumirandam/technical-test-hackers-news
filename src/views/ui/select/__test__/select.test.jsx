/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen, fireEvent } from '@utils/test-utils';

import Select from '../index';

const defaultProps = {
  options: [{ value: 'reactjs', label: 'Reactjs' }],
};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Select {...setupStore}>Press</Select>, {});
};

describe('Test for Select ui component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('Reactjs')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render change select', () => {
    setup();
    const select = screen.getByTestId('Select');
    fireEvent.change(select, {
      target: { value: 'reactjs' },
    });
    expect(screen.getByText('Reactjs')).toBeTruthy();
  });
});
