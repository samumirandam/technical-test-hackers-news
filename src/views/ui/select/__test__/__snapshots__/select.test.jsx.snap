// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Select ui component Should renders the same component 1`] = `
<div>
  <select
    class="Select"
    data-testid="Select"
  >
    <option
      value="reactjs"
    >
      Reactjs
    </option>
  </select>
</div>
`;
